sdk/core/Azure.Core/src/ETag.cs

```diff
@@ -41,5 +45,24 @@ public override string ToString()
         {
             return _value ?? "<null>";
         }
+
+        internal static ETag Parse(string stringValue)
+        {
+            if (stringValue == All._value)
+            {
+                return All;
+            }
+            else if (stringValue.StartsWith("W/", StringComparison.OrdinalIgnoreCase))
+            {
+                throw new NotSupportedException("Weak ETags are not supported.");
+            }
+            else if (!stringValue.StartsWith(QuoteString, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
A quote doesn't have case-sensitivity, and using `OrdinalIgnoreCase` will have perf ramifications unnecessarily.

Reply:
Good point!